### PR TITLE
Fix logic with OR operation and add missing nil check to Like

### PR DESCRIPTION
**Pull request by VinaiRachakonda**
MySQL has some really weird behavior where the OR of two non bools evaluates too false. Improperly handling this situation was trickling up to a LIKE panic on the new query added to queries.go

```
SELECT "A" OR "A";
+------------+
| "A" OR "A" |
+------------+
|          0 |
+------------+
```

### CODE REVIEW

**Review by jennifersp — APPROVED** [review]
LGTM!

File: enginetest/queries.go [hunk 1]
```diff
@@ -6738,6 +6738,10 @@ var QueryTests = []QueryTest{
 			{1, 1, 0},
 		},
 	},
+	{
+		Query:    `select (('%' || 'dsads') || '%')`,
```
Comment (zachmu) [inline enginetest/queries.go:6742]: Use `OR` instead of `||`

File: sql/expression/like.go [hunk 2]
```diff
@@ -110,6 +110,10 @@ func (l *Like) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		return nil, err
 	}
 
+	if likeMatcher == nil {
```
Comment (zachmu) [inline sql/expression/like.go:113]: Where's the test for this fix?

**Review by zachmu — COMMENTED** [review]
LG but missing a test

File: sql/expression/like.go (hunk 2 above)
Comment (VinaiRachakonda) [reply sql/expression/like.go:113]: The one in queries.go triggered this originally but I can add more.

File: enginetest/queries.go (hunk 1 above)
Comment (VinaiRachakonda) [reply enginetest/queries.go:6750]: Whoops will fix

**Review by zachmu — APPROVED** [review]
LGTM

**Pull request merged**